app/lib/features/link_room/widgets/link_room_trailing.dart

```diff
@@ -0,0 +1,177 @@
+    if (joinRule == 'restricted' || joinRule == 'knock_restricted') {
+      currentRooms = asDartStringList(room.restrictedRoomIdsStr());
+      parentCanSee = currentRooms.contains(parentSpaceId);
+      newRule = joinRule;
+    }
+
+    if (!parentCanSee) {
+      final spaceAvatarInfo = ref.read(roomAvatarInfoProvider(parentSpaceId));
+      final parentSpaceName =
+          spaceAvatarInfo.displayName ?? lang.theParentSpace;
+      final roomName =
+          // ignore: use_build_context_synchronously
+          spaceAvatarInfo.displayName ?? lang.theSelectedRooms;
+      bool shouldChange = await showDialog(
+        context: context,
+        builder: (context) {
+          return AlertDialog(
+            title: Text(lang.notVisible),
+            content: Wrap(
+              children: [
+                Text(
+                  lang.theCurrentJoinRulesOfSpace(roomName, parentSpaceName),
+                ),
+              ],
+            ),
+            actionsAlignment: MainAxisAlignment.spaceEvenly,
+            actions: <Widget>[
+              OutlinedButton(
+                key: LinkRoomPage.denyJoinRuleUpdateKey,
+                child: Text(lang.noThanks),
+                onPressed: () {
+                  Navigator.pop(context, false);
+                },
+              ),
+              ActerPrimaryActionButton(
+                key: LinkRoomPage.confirmJoinRuleUpdateKey,
+                child: Text(lang.yesPleaseUpdate),
+                onPressed: () {
+                  Navigator.pop(context, true);
+                },
+              ),
+            ],
+          );
+        },
+      );
+      if (shouldChange) {
+        currentRooms.add(parentSpaceId);
+
+        final sdk = await ref.read(sdkProvider.future);
+        final update = sdk.api.newJoinRuleBuilder();
+        update.joinRule(newRule);
+        for (final roomId in currentRooms) {
+          update.addRoom(roomId);
+        }
+        await room.setJoinRule(update);
+      }
+    }
+  }
+
+//Link child room
+  Future<void> onTapLinkChildRoom(BuildContext context, WidgetRef ref) async {
+    final lang = L10n.of(context);
+    final room = await ref.read(maybeRoomProvider(roomId).future);
+    if (room == null) {
+      EasyLoading.showError(lang.roomNotFound);
+      return;
+    }
+
+    final canLink = (await ref.read(roomMembershipProvider(roomId).future))
+            ?.canString('CanLinkSpaces') ==
+        true;
+
+    //Fetch selected parent space data and add given roomId as child
+    final space = await ref.read(spaceProvider(parentId).future);
+    await space.addChildRoom(roomId, false);
+
+    //Make subspace
+    if (canLink) {
+      //Fetch selected room data and add given parentSpaceId as parent
+      await room.addParentRoom(parentId, true);
+      if (!context.mounted) return;
+      await checkJoinRule(context, ref, room, parentId);
+    } else {
+      EasyLoading.showSuccess(lang.roomLinkedButNotUpgraded);
+    }
```

Comment:
This now accepts both the non upgradable and the uprgadable cases and only shows the dialog if you can upgrade. If you can't uprgade you'll get a success messages with a note instead.